### DISCUSSION TITLE
GeoFencing Sample

### DISCUSSION THREAD

**Discussion by BicycleMark** [shinyorg/shiny#999] [Q&A] [answered]
Is Shiny a good GeoFencing Solution? I see that it has a GeoFencing Sample....

Here it is:
[](https://github.com/shinyorg/samples/tree/main/Locations-Geofencing)

The GeoFencing Sample does not build. From looks of it it has not been building for a while now. 

If it were simple to fix this sample it would have been done by now in the OSS community. I just gave it 2 hours ran in to a dead end

How to use GeoFencing? Is Shiny all set and ready to do GeoFencing?  Is there another sample somewhere that uses Shiny and GeoFencing?

Kind Regards,
Mark Wardell

**Comment by aritchie** [comment]
> Is Shiny a good GeoFencing Solution? I see that it has a GeoFencing Sample....

Shiny works perfectly for geofencing.  I use it in a bunch of projects.

> The GeoFencing Sample does not build. From looks of it it has not been building for a while now.

I just opened the geofence sample & compiled it just fine.

> If it were simple to fix this sample it would have been done by now in the OSS community. I just gave it 2 hours ran in to a dead end

No.  People come in with an attitude and expect something for free.... with a very similar tone to your own.  

Perhaps you can explain the issue(s) you are having and I'll see if I can point you in a direction?  However, if you expect to get help, please change your tone a bit as you coming off rude.

**Comment by BicycleMark** [comment]
It may be very simple - The samples repo references a folder
.*./shiny2/ *outside
of the repo.

Here are the steps to repeat:

1) create a folder on Mac ~/Documents/src2
2) cd to ~Documents/src2
2) open terminal perform a "git clone
https://github.com/shinyorg/samples.git"
3) open the sln in question (Geofencing.sln) - the sln references a folder
that is not part of the repo. I am thinking it is a bad checkin. Maybe
something odd like I really needed to have a dependent repo carefully
placed in a folder before I issue 
![image](https://user-images.githubusercontent.com/11739690/167966763-057022d1-0edd-4e8d-ae71-eb8f65e975e2.png)

Apparently the csproj references *shiny2* not part of the clone is missing
some reference you have on your hard drive. Was I supposed to download some
other repo first? "git clone https://github.com/shinyorg/samples.git"

FYI The Build on Shiny components also indicates that CI is struggling to
build also. It seems to be flashing red.

<img width="479" alt="image" src="https://user-images.githubusercontent.com/11739690/167966812-cbc59c67-6646-439a-9d4b-52ac7cb6d825.png">

**Reply by BicycleMark** [reply]
I also tried --recurse-submodules like so:
git clone --recurse-submodules https://github.com/shinyorg/samples.git

No luck :( I came last fall and this was in same state at that point.

**Reply by aritchie** [reply]
Please read the readme on the samples repo - the very first statement says to ignore missing projects.  I put shared references in the samples for when I need to debug.  They aren't actually referenced by the samples unless you change the props file.

> FYI The Build on Shiny components also indicates that CI is struggling to build also. It seems to be flashing red.

Nothing is struggling to build.  Master is part of the alpha.  Only the v(branches) matter.

**Reply by BicycleMark** [reply]
Okay then! We can all agree that a 1/2 dozen error on open of a solution is  a very very  weird way to introduce yourself to the world. Your world?  Your weird rules. Thanks!

**Reply by aritchie** [reply]
Wow... what a horrible entitled attitude you have.  Welcome to the ban world :)

**Comment by roubachof** [comment, marked as answer]
<img width="874" alt="image" src="https://user-images.githubusercontent.com/596903/167999697-9a632436-2ece-4c56-942c-55bbd07648c3.png">

Wait! ARE WE SUPPOSED TO READ THE README????
What an outrageous answer by @aritchie, this is unacceptable!
